Fails to compile string builder operation

### Describe the bug

```
/tmp/v_501/bug_test.2391090100100521907.tmp.c:19047:4: warning: expression result unused [-Wunused-value]
  (*(string*)_t1.data);
   ^~~~~~~~~~~~~~~~~~
/tmp/v_501/bug_test.2391090100100521907.tmp.c:22494:52: error: initializing 'u8' (aka 'unsigned char') with an expression of incompatible type 'array' (aka 'struct array')
                        array_push_noscan((array*)&result, _MOV((u8[]){ array_reverse(sub_result) }));
                                                                        ^~~~~~~~~~~~~~~~~~~~~~~~~
1 warning and 1 error generated.
builder error: 
==================
C error. This should never happen.
```

### Expected Behavior

Successful compilation or type error. ;-)

### Current Behavior

Fails to compile.

### Reproduction Steps

```vlang
module bug

import rand
import strings

fn test_trigger_bug() {
	mut a := strings.new_builder(0)
	mut b := strings.new_builder(0)
	b << rand.ascii(5).bytes()
	a << a.reverse()
}
```


### Possible Solution

_No response_

### Additional Information/Context

_No response_

### V version

V 0.3.3 c6947fd

### Environment details (OS name and version, etc.)

V full version: V 0.3.3 838083e.c6947fd
OS: macos, macOS, 13.2.1, 22D68
Processor: 12 cpus, 64bit, little endian, Apple M2 Max

getwd: /Users/ta/PycharmProjects/4917
vexe: /Users/ta/.v/v
vexe mtime: 2023-04-13 13:01:09

vroot: OK, value: /Users/ta/.v
VMODULES: OK, value: /Users/ta/.vmodules
VTMP: OK, value: /tmp/v_501

Git version: git version 2.37.1 (Apple Git-137.1)
Git vroot status: weekly.2023.15-10-gc6947fde
.git/config present: true

CC version: Apple clang version 14.0.0 (clang-1400.0.29.202)
thirdparty/tcc status: thirdparty-macos-arm64 a668e5a0

--- a/vlib/v/gen/c/infix.v
+++ b/vlib/v/gen/c/infix.v
@@ -741,8 +741,9 @@ fn (mut g Gen) infix_expr_left_shift_op(node ast.InfixExpr) {
 		tmp_var := g.new_tmp_var()
 		array_info := left.unaliased_sym.info as ast.Array
 		noscan := g.check_noscan(array_info.elem_type)
-		if right.unaliased_sym.kind == .array && array_info.elem_type != right.typ
-			&& !(right.sym.kind == .alias
+		if (right.unaliased_sym.kind == .array
+			|| (right.unaliased_sym.kind == .struct_ && right.unaliased_sym.name == 'array'))
+			&& array_info.elem_type != right.typ && !(right.sym.kind == .alias
 			&& g.table.sumtype_has_variant(array_info.elem_type, node.right_type, false)) {
 			// push an array => PUSH_MANY, but not if pushing an array to 2d array (`[][]int << []int`)
 			g.write('_PUSH_MANY${noscan}(')

--- a/vlib/v/tests/strings_builder_shift_array_reverse_test.v
+++ b/vlib/v/tests/strings_builder_shift_array_reverse_test.v
@@ -1,0 +1,12 @@
+import rand
+import strings
+
+fn test_strings_builder_shift_array_reverse() {
+	mut a := strings.new_builder(0)
+	mut b := strings.new_builder(0)
+	b << rand.ascii(5).bytes()
+	a << b.reverse()
+	ret := a.str()
+	println(ret)
+	assert ret.len == 5
+}

